server/resp/src/test/java/org/infinispan/server/resp/JsonCommandsTest.java

```diff
@@ -0,0 +1,462 @@
+package org.infinispan.server.resp;
+
+import static org.assertj.core.api.Assertions.assertThat;
+import static org.assertj.core.api.Assertions.assertThatThrownBy;
+import static org.infinispan.server.resp.test.RespTestingUtil.assertWrongType;
+import static org.infinispan.test.TestingUtil.k;
+import static org.infinispan.test.TestingUtil.v;
+
+import java.util.List;
+
+import org.infinispan.server.resp.json.JSONUtil;
+import org.testng.annotations.BeforeMethod;
+import org.testng.annotations.Test;
+
+import com.fasterxml.jackson.databind.JsonNode;
+import com.fasterxml.jackson.databind.ObjectMapper;
+import com.fasterxml.jackson.databind.node.ArrayNode;
+import com.fasterxml.jackson.databind.node.ObjectNode;
+import com.jayway.jsonpath.Configuration;
+
+import io.lettuce.core.RedisCommandExecutionException;
+import io.lettuce.core.api.sync.RedisCommands;
+import io.lettuce.core.json.DefaultJsonParser;
+import io.lettuce.core.json.JsonArray;
+import io.lettuce.core.json.JsonPath;
+import io.lettuce.core.json.JsonValue;
+import io.lettuce.core.json.arguments.JsonGetArgs;
+import io.lettuce.core.json.arguments.JsonSetArgs;
+
+@Test(groups = "functional", testName = "server.resp.JsonCommandsTest")
+public class JsonCommandsTest extends SingleNodeRespBaseTest {
+   /**
+    * RESP Sorted set commands testing
+    *
+    * @since 15.1
+    */
+   RedisCommands<String, String> redis;
+   private Configuration config = JSONUtil.config;
+
+   @BeforeMethod
+   public void initConnection() {
+      redis = redisConnection.sync();
+   }
+
+   @Test
+   public void testJSONSET() {
+      String key = k();
+      JsonPath jp = new JsonPath("$");
+      JsonValue jv = new DefaultJsonParser().createJsonValue("{\"key\":\"value\"}");
```

Comment:
I see the `new DefaultJsonParser()` is instantiated every test, sometimes more than once. Maybe this should be an instance field.